torch/nested/_internal/ops.py

```diff
@@ -2170,16 +2195,40 @@ def to_padded_tensor_default(func, *args, **kwargs):
     return padded_out
 
 
-@register_jagged_func(
-    torch.ops.aten._nested_from_padded_tensor.default,
-    "padded: t, offsets: t, dummy: jt, ragged_idx: any?, min_seqlen: any?, max_seqlen: any?, sum_S: any?",
-)
+from torch.nested._internal.cached_tensor import register_cached_tensor_func
+
+
+# When we use [ CacheTensor open registry ], we cannot rely on NestedTensor's
+# torch function anymore to enable maybe_enable_thunkify. All functions
+# registered to register_cached_tensor_func must be first wrapped with this
+# decorator.
+def maybe_enable_thunkify(f):
+    import functools
+
+    from torch.fx.experimental.proxy_tensor import maybe_enable_thunkify
+
+    @functools.wraps(f)
+    def fn(*args, **kwargs):
+        with maybe_enable_thunkify():
+            return f(*args, **kwargs)
+
+    return fn
+
+
+# See Note: [ CacheTensor open registry ]
+@register_cached_tensor_func(torch.ops.aten._nested_from_padded_tensor.default)
+@maybe_enable_thunkify
```

Comment:
do we need the `maybe_enable_thunkify`? 🤔

Reply:
So the context is that @ezyang had a change to disable thunkifying by default due to the recursion being too deep. https://github.com/pytorch/pytorch/pull/132421  Disabling thunkify means symint operations are eagerly placed into the graph. 

My wishy washy understanding of the issue here is that there are some problematic symint operations involving NJT that previously would've died, but cause problems. I haven't looked further though.

Reply:
Yeah, if you just remove the maybe enable thunkify you will find out why it fails :P But some of this was blocked on the work on making dynamic shapes + subclasses work correctly (did @guilhermeleobas land all of that?)

Reply:
Yes! But there was another issue that prevent us to remove `maybe_enable_trunkify`